Objects/longobject.c

```diff
@@ -6685,3 +6685,47 @@ Py_ssize_t
 PyUnstable_Long_CompactValue(const PyLongObject* op) {
     return _PyLong_CompactValue((PyLongObject*)op);
 }
+
+const PyUnstable_LongLayout PyUnstable_Long_LAYOUT = {
+    .bits_per_digit = PyLong_SHIFT,
+    .word_endian = PY_LITTLE_ENDIAN ? -1 : 1,
+    .array_endian = -1,  // least significant first
+    .digit_size = sizeof(digit),
+};
+
+
+PyObject*
+PyUnstable_Long_Import(int negative, size_t ndigits, Py_digit *digits)
+{
+    return (PyObject*)_PyLong_FromDigits(negative, ndigits, digits);
```

Comment:
Hmm, there is a bug, either here or in the _PyLong_FromDigits().  That functions has no special case for small integers (cached), except for zero.  Thus, this will create them, instead of taking a new reference.

Code in the decimal module, using this API, does special case for 1-digit int's.  I suggest to do same:
```suggestion
    if (ndigits == 1) {
        return PyLong_FromLong((negative?-1:1)*digits[0]);
    }
    return (PyObject*)_PyLong_FromDigits(negative, ndigits, digits);
```

Reply:
Fixed in the reimplementation of `_PyLong_FromDigits()`, see below.

Reply:
Yes, but please note use cases for this function in the CPython.  Your additions should slightly increase overhead (writer allocation, checks, etc).

Reply:
I'm first trying to write a correct implementation and discuss the API. We can talk about micro-optimization later :-) (I can revert my _PyLong_FromDigits() changes later if the overhead is not acceptable.)